v2/proto/claim.proto

```diff
@@ -11,53 +11,736 @@ message Claim {
 
 message Channel {
     bytes public_key = 1;
+
     string title = 2;
     string description = 3;
-    string contact_email = 4;
-    string homepage_url = 5;
-    repeated string tags = 6;
+    string thumbnail_url = 4;
+    repeated string tags = 5;
+    repeated Language languages = 6;
+    repeated Location locations = 7;
 
-    string thumbnail_url = 16;
-    string cover_url = 17;
-    string language = 18;
+    string contact_email = 8;
+    string homepage_url = 9;
+    string cover_url = 10;
 }
 
 message Stream {
-    bytes hash = 1;
-    string language = 2;
-    string title = 3;
-    string author = 4;
-    string description = 5;
-    string media_type = 6;
-    string license = 7;
-    File file = 8;
-
-    Fee fee = 16;
-    string license_url = 17;
-    string thumbnail_url = 18;
-    uint32 duration = 19;
-    repeated string tags = 20;
-    int64 release_time = 21; // seconds since UNIX epoch
-    Video video = 22;
+    bytes sd_hash = 1;
+
+    string title = 2;
+    string description = 3;
+    string thumbnail_url = 4;
+    repeated string tags = 5;
+    repeated Language languages = 6;
+    repeated Location locations = 7;
+
+    string author = 8;
+    string license = 9;
+    string license_url = 10;
+    int64 release_time = 11; // seconds since UNIX epoch
+
+    string media_type = 13;
+    File file = 14;
+    Fee fee = 15;
+
+    oneof type {
+      Image image = 16;
+      Video video = 17;
+      Audio audio = 18;
+   }
 }
 
 message Fee {
     enum Currency {
-        LBC = 0;
-        USD = 1;
+        UNKNOWN_CURRENCY = 0;
+        LBC = 1;
+        BTC = 2;
+        USD = 3;
     }
-
     Currency currency = 1;
     bytes address = 2;
-    uint64 amount = 3; // deweys for LBC, cents for USD
+    uint64 amount = 3; // deweys for LBC/BTC, cents for USD
 }
 
 message File {
+    bytes hash = 3;
     string name = 1;
     uint64 size = 2;
 }
 
+message Image {
+    uint32 width = 1;
+    uint32 height = 2;
+}
+
 message Video {
     uint32 width = 1;
     uint32 height = 2;
+    uint32 duration = 3;
+}
+
+message Audio {
+    uint32 duration = 1;
+}
+
+// RFC 5646
+message Language {
+    Language language = 1;
+    Script script = 2;
+    Location.Country region = 3;
+
+    // ISO 639-1
+    enum Language {
+        UNKNOWN_LANGUAGE = 0;
```

Comment:
1 - all these `UNKNOWN_*` fields don't make sense. if its unknown, just leave it out
2 - common languages should go into the top 15. helps reduce the size.